### DISCUSSION TITLE
In a JavaScript action, how do I execute a Git command and save the output to a variable?

### DISCUSSION THREAD

**Discussion by cagross** [community/community#26546] [Actions]
<p>Hello.  I have created a JavaScript GitHub action, and inside, want to execute a simple Git command and save it to a JavaScript variable.  Using the <a href="https://github.com/actions/toolkit/tree/HEAD/packages/exec" rel="noopener nofollow ugc">exec toolkit package</a>, I can execute the Git command.  That prints the correct output to the console (<code>06aee1b</code>).  But when I try to save the output to a variable, the value of the variable is <code>0</code>.  How do I save the correct output to a variable?  My code and output are below:</p>
<p>I have intermediate JavaScript and Git experience.  But this is my first experience with GitHub actions.</p>
<hr>
<p><strong>My Code (in my JavaScript action)</strong></p>
<pre><code class="lang-javascript">import * as core from "@actions/core";
import * as github from "@actions/github";
import * as exec from "@actions/exec";

const execOutput = await exec.exec("git rev-parse --short HEAD");
console.log("execOutput is: ");
console.log(execOutput);
</code></pre>
<p><strong>Output</strong></p>
<pre><code class="lang-auto">06aee1b
execOutput is: 
0
</code></pre>

**Comment by laughedelic** [comment, marked as answer]
<p>Check this part of the <a href="https://github.com/actions/toolkit/tree/HEAD/packages/exec#outputoptions" rel="noopener nofollow ugc">docs</a>:</p>
<pre><code class="lang-auto">const exec = require('@actions/exec');

let myOutput = '';
let myError = '';

const options = {
  listeners: {
    stdout: (data: Buffer) =&gt; {
      myOutput += data.toString();
    },
    stderr: (data: Buffer) =&gt; {
      myError += data.toString();
    }
  }
};

await exec.exec('git', ['rev-parse', '--short', 'HEAD'], options);
console.log(myOutput);
console.log(myError);
</code></pre>

**Comment by cagross** [comment]
<p>OK thanks so much–that’s exactly what I needed.  Didn’t read the docs carefully enough :-/</p>
